### ISSUE TITLE
Closing the NATS instance with no connection established throws NatsError: Subject must be supplied

### ISSUE THREAD

**Issue by anXieTyPB** [nats-io/node-nats-streaming#92]
When attempting to close a node-nats-streaming instance that is trying to connect to a cluster because waitOnFirstConnect is set to true, it throws an "NatsError: Subject must be supplied" error every time.

Attached is a script that showcases the problem. Am I using close in an incorrect way? If yes, what is the best practice of closing a nats connection that might not be connected yet because the application is closing?

```
// docker run -p 4222:4222 -p 8223:8223 nats-streaming
const natsstreaming = require('node-nats-streaming')
const host = `nats://192.168.99.100:4224`
const opts = {
  url: host,
  maxReconnectAttempts: 12,
  reconnect: true,
  reconnectTimeWait: 6000,
  verbose: true,
  waitOnFirstConnect: true
}

const stan = natsstreaming.connect('test-cluster', `some client id`, opts)

stan.on('connect', () => {
  console.log('NATS connection established.')
})

stan.on('reconnect', () => {
  console.log('NATS reconnected.')
})

stan.on('reconnecting', () => {
  console.log('NATS reconnecting...')
})

stan.on('disconnect', () => {
  console.log('NATS disconnected.')
})

stan.on('error', (err) => {
  console.error(err)
})

stan.on('close', () => {
  console.log('NATS connection closed.')
})

process.on('SIGINT', () => {
  console.log('Received SIGINT, closing NATS instance')
  stan.close()
})

process.on('SIGTERM', () => {
  console.log('Received SIGTERM, closing NATS instance')
  stan.close()
})

setTimeout(() => {
  console.log('attempting to close nats')
  stan.close()
}, 2500)

```
Output when application is run without further user interaction:
```
NATS disconnected.
attempting to close nats

[...]\node_modules\nats\lib\nats.js:1379
            throw (new NatsError(BAD_SUBJECT_MSG, BAD_SUBJECT));
            ^
NatsError: Subject must be supplied
    at Client.publish ([...]node_modules\nats\lib\nats.js:1379:20)
    at Client.request ([...]node_modules\nats\lib\nats.js:1606:10)
    at Client.requestOne ([...]node_modules\nats\lib\nats.js:1685:17)
    at Stan.close ([...]node_modules\node-nats-streaming\lib\stan.js:364:11)
    at Timeout.setTimeout [as _onTimeout] 
    [...]
```

You can also remove the setTimeout code block and issue a SIGTERM at some point in time. Output: 
```
NATS disconnected.
NATS reconnecting...
NATS disconnected.
NATS reconnecting...
NATS disconnected.
NATS reconnecting...
NATS disconnected.
Received SIGINT, closing NATS instance

[...]node_modules\nats\lib\nats.js:1379
            throw (new NatsError(BAD_SUBJECT_MSG, BAD_SUBJECT));
            ^
NatsError: Subject must be supplied
    at Client.publish ([...]node_modules\nats\lib\nats.js:1379:20)
    at Client.request ([...]node_modules\nats\lib\nats.js:1606:10)
    at Client.requestOne [...]node_modules\nats\lib\nats.js:1685:17)
    at Stan.close ([...]node_modules\node-nats-streaming\lib\stan.js:364:11)
    at process.on ([...]\connection_natsonly.js:42:8)
    at process.emit (events.js:197:13)
```

Edit:
node-nats-streaming version: 0.0.52

**Comment by aricart** [comment]
The STAN protocol requires for the stan connection to form before any of the STAN API is called.
@aknuds1 thanks for the report.

**Comment by aricart** [comment]
Fixed by #95
